packages/host/src/directory.toit

```diff
@@ -95,17 +95,17 @@ class DirectoryStream:
   */
   next -> string?:
     while true:
-      byte_array := readdir_ fd_
+      byte_array := readdir_ dir_
       if not byte_array: return null
       str := byte_array.to_string
       if str == "." or str == "..": continue
       return str
 
   close -> none:
-    fd := fd_
-    closedir_ fd
+    dir := dir_
```

Comment:
What's up with this dance? Should we set dir_ to null here?

Reply:
That depends on whether we want to silently allow multiple close calls.  I think in general that's just nicer.  If we leave it as is the underlying Directory is deleted, but the proxy lives on with a null pointer in it.  That means we get an ALREADY_CLOSED exception if we call close twice.